arangod/VocBase/LogicalCollection.cpp

```diff
@@ -387,34 +393,47 @@ uint64_t LogicalCollection::numberDocuments(transaction::Methods* trx,
 }
 
 bool LogicalCollection::hasClusterWideUniqueRevs() const noexcept {
+  TRI_ASSERT(_sharding != nullptr);
   return usesRevisionsAsDocumentIds() && isSmartChild();
 }
 
 bool LogicalCollection::mustCreateKeyOnCoordinator() const noexcept {
+  TRI_ASSERT(_sharding != nullptr);
   TRI_ASSERT(ServerState::instance()->isRunningInCluster());
   // when there is more than 1 shard, or if we do have a satellite
   // collection, we need to create the key on the coordinator.
   return numberOfShards() != 1;
 }
 
-uint32_t LogicalCollection::v8CacheVersion() const { return _v8CacheVersion; }
+uint32_t LogicalCollection::v8CacheVersion() const {
+  TRI_ASSERT(_sharding != nullptr);
+  return _v8CacheVersion;
+}
 
-TRI_col_type_e LogicalCollection::type() const { return _type; }
+TRI_col_type_e LogicalCollection::type() const {
+  return _type;
+}
 
-TRI_vocbase_col_status_e LogicalCollection::status() const { return _status; }
+TRI_vocbase_col_status_e LogicalCollection::status() const {
+  TRI_ASSERT(_sharding != nullptr);
+  return _status;
+}
 
 TRI_vocbase_col_status_e LogicalCollection::getStatusLocked() {
+  TRI_ASSERT(_sharding != nullptr);
```

Comment:
Are all of those new asserts really required here? Seems to be added to every method:
```
  TRI_ASSERT(_sharding != nullptr);
```

Reply:
Michael's decision.

Reply:
Uh I think we had a misunderstanding here.
is the _sharding is accessed within the method it shall be asserted.
If not, it shall not be asserted. (it is not required.)